cdap-ui/server/express.js

```diff
@@ -471,6 +471,47 @@ function makeApp(authAddress, cdapConfig, uiSettings) {
     },
   ]);
 
+  app.all('/api/:path(*)', function(req, res) {
+    var protocol, port;
+    if (cdapConfig['ssl.external.enabled'] === 'true') {
+      protocol = 'https://';
+    } else {
+      protocol = 'http://';
+    }
+    if (cdapConfig['ssl.external.enabled'] === 'true') {
+      port = cdapConfig['router.ssl.server.port'];
+    } else {
+      port = cdapConfig['router.server.port'];
+    }
+
+    var url = [
+      protocol,
+      cdapConfig['router.server.address'],
+      ':',
+      port,
+      '/',
+      req.param('path'),
+    ].join('');
+    var opts = {
+      method: req.method,
+      url: url,
+      headers: req.headers,
+    };
+
+    req
+      .on('error', function(e) {
+        log.error(e);
+      })
+      .pipe(request(opts))
+      .on('error', function(e) {
+        log.error(e);
+      })
+      .pipe(res)
+      .on('error', function(e) {
+        log.error(e);
```

Comment:
send errors to the response with `500` error?

Reply:
This is already done in the pipe to res ? This is here only for logging purposes right?

Reply:
sorry, yeah. Disregard comment